csunplugged/templates/dev/index.html

```diff
@@ -7,70 +7,95 @@
+              <li>Unit Plan: <a href="{% url 'topics:unit_plan' topic.slug unit_plan.slug %}">{{ unit_plan.name }}</a></li>
+
+              {% if unit_plan.lessons.items %}
+                <ul>
+                  {% for age_group,lessons in unit_plan.lessons.items %}
+                    <li>Lessons ({{ age_group.0 }}-{{ age_group.1}}):</li>
+                    <ul>
+                      {% for lesson in lessons %}
+                        <li><a href="{% url 'topics:lesson' topic.slug unit_plan.slug lesson.slug %}">{{ lesson }}</a></li>
+                      {% endfor %}
+                    </ul>
+                  {% endfor %}
+                </ul>
+              {% endif %}
+
             {% endfor %}
-          </ul>
-        {% endfor %}
-        <li>Curriculum integrations:</li>
-        <ul>
-          {% for integration in topic.integrations %}
-            <li><a href="{% url 'topics:integration' topic.slug integration.slug %}">{{ integration }}</a></li>
-          {% endfor %}
+          {% endif %}
+
+          {% if topic.integrations.all %}
+            <li>Curriculum integrations:</li>
+            <ul>
+              {% for integration in topic.integrations %}
+                <li><a href="{% url 'topics:integration' topic.slug integration.slug %}">{{ integration }}</a></li>
+              {% endfor %}
+            </ul>
+          {% endif  %}
+
+          {% if topic.programming_exercises.all %}
+            <li>Programming Exercises:</li>
+            <ul>
+              {% for exercise in topic.programming_exercises %}
+                <li><a href="{% url 'dev:programming_exercise' exercise.slug %}">{{ exercise.exercise_set_number }}.{{ exercise.exercise_number }} {{ exercise.name }}</a></li>
+              {% endfor %}
+            </ul></li>
+          {% endif %}
+
         </ul>
-        <li>Programming Exercises:</li>
+      {% endfor %}
+    </ul>
+  {% endif %}
+
+  {% if curriculum_areas %}
+    <h1>Curriculum Areas</h1>
+    <ul>
+      {% for parent,children in curriculum_areas.items %}
+        <li>{{ parent.name }}</li>
         <ul>
-          {% for exercise in topic.programming_exercises %}
-            <li><a href="{% url 'dev:programming_exercise' exercise.slug %}">{{ exercise.exercise_set_number }}.{{ exercise.exercise_number }} {{ exercise.name }}</a></li>
-          {% endfor %}
-        </ul></li>
-      </ul>
-    {% endfor %}
-  </ul>
-
-  <h1>Curriculum Areas</h1>
-  <ul>
-    {% for parent,children in curriculum_areas.items %}
-      <li>{{ parent.name }}</li>
-      <ul>
-      {% for child in children %}
-        <li>{{ child.name }}</li>
+        {% for child in children %}
+          <li>{{ child.name }}</li>
+        {% endfor %}
+        </ul>
+      {% endfor %}
+    </ul>
+  {% endif %}
+
+  {% if learning_outcomes.all %}
+    <h1>Learning Outcomes</h1>
+    <ul>
+      {% for learning_outcome in learning_outcomes.all %}
+        <li>{{ learning_outcome }}</li>
```

Comment:
It would be good to display both the key and text for each learning outcome. Because if an author finds an outcome here they wish to use, they then have to find it in the YAML file.